### ISSUE TITLE
RFE - provide auto tuning/sizing profiles

### ISSUE THREAD

**Issue by mreynolds389** [389ds/389-ds-base#5757]
Currently our auto sizing for threads, which was recently changed to use less threads per core, is adversely affecting some customers with different load patterns.  

The cache auto sizing default is also way too conservative.  Part of the issue is that customers think auto tuning is maximizing the tuning, while it is actually just an improved default from years ago.  So they don't always look at tuning because they think they are already at the best tuning possible.

We could benefit from providing different "sizing profiles" that you can select (or create your own).  Some being aggressive, while others not, etc.  This will need more investigation to identify these "profiles", but we can definitely improve things here and make it more intuitive/easy for admins to tune the server for different types of load (load profiles).

**Comment by Firstyear** [comment]
The memory tuning had to be conservative by default because of freeipa where we are not the only thing running (especially given the large footprint of other parts of the freeipa stack). :(

**Comment by mreynolds389** [comment]
> The memory tuning had to be conservative by default because of freeipa where we are not the only thing running (especially given the large footprint of other parts of the freeipa stack). :(

I know, I was not trying to be insulting.  I just think it's time to evolve the feature a little bit

**Comment by Firstyear** [comment]
More that I think we have to leave the default as quite paranoid. I think the profiles are a good idea though. 
:)

**Comment by mreynolds389** [comment]
> More that I think we have to leave the default as quite paranoid. I think the profiles are a good idea though. :)

Exactly. I was thinking of having the default profile be the current auto sizing settings.  Then we can have other profiles for more aggressive settings, or mixed & matched settings.   We know of some new scenarios where thread tuning needs to be different, but there are other scenarios that we can incorporate.  

We have been in touch with a consultant who has been dealing with these auto tuning issues (where it needs to be different depending on the type of load, etc).  So we need to continue to look into these situations where it makes sense to use different settings.  It will take some time, but adding profiles down the road is trivial.
